### DISCUSSION TITLE
"Missing name" @ MUI TextField with useFieldArray

### DISCUSSION THREAD

**Discussion by dandv** [react-hook-form/react-hook-form#1693] [General] [answered]
@bluebill1049: thanks for solving the `register()` problem in the field array discussion! I'm trying to use that sample with MUI TextField in [this CSB](https://codesandbox.io/s/react-hook-form-usefieldarray-error-on-removal-qz9iw?file=/src/DeletableField.js) but I see a `"Missing name @ <div ...>"` warning, where the div is output by the TextField.

Does RHF need the `name` property on the root of each fieldArray element, even if that element is not the actual `<input>`? The `TextField` passes `name` down to the `<input>` and I couldn't find a way to make it keep the `name` property on itself. `InputProps={{name}}` didn't help, and that would be the inner `OutlinedInput` (another `<div>`) anyway.

**Comment by bluebill1049** [comment, marked as answer]
I would use Controller instead for the external controlled component: https://react-hook-form.com/api#Controller

```
<TextField
  ref={register({ name })}
  x-id="TextField"  // to show in the "Missing name @" warning
  variant="outlined"
  {...others}
/>
```

**Reply by dandv** [reply]
Thanks, and also thanks for [all the MUI examples](https://codesandbox.io/s/react-hook-form-controller-079xx)!
